utils/build_swift/build_swift/shell.py

```diff
@@ -249,11 +227,11 @@ def quote(command):
     "rm -rf '~/Documents/My Homework'"
     """
 
-    if isinstance(command, six.string_types):
-        return _quote(command)
+    if isinstance(command, str):
+        return shlex.quote(command)
```

Comment:
This is a problem - `shlex` is only applicable to Unix platforms.

Reply:
This doesn't actually change the compatibility. As per the above import lines, the `_quote` function was coming from `pipes` before. `shutil.quote` doesn't exist, but `pipes.quote` did (although it was undocumented).

From the `pipes` documentation in python2.7:

> Because the module uses `/bin/sh` command lines, a POSIX or compatible shell for `os.system()` and `os.popen()` is required.

And from the `pipes.quote(s)` documentation in python2.7:

> _Deprecated since version 2.7_: Prior to Python 2.7, this function was not publicly documented. It is finally exposed publicly in Python 3.3 as the `quote` function in the `shlex` module.

If we _do_ need a cross-platform shell quoting function, I don't know of one built-in.

Reply:
Hmm, the undocumented version I thought did both ... but the shlex is documented as being Unix only.  This is a problem .. but I guess this change doesn't make it any worse so its fine.  This is part of the problem that requires the custom build rules for WIndows.

Reply:
Looking at the stdlib code, `pipes.quote` is literally just an alias for `shlex.quote`, so if it worked before, it will work now.

From pipes.py:63-65 ::
```py
# we import the quote function rather than the module for backward compat
# (quote used to be an undocumented but used function in pipes)
from shlex import quote
```